src/components/day-card-list/day-card-list.tsx

```diff
@@ -1,7 +1,64 @@
-import type { Component } from "solid-js";
+import {
+  Component,
+  createEffect,
+  createSelector,
+  createSignal,
+  Show,
+} from 'solid-js';
+import { For } from 'solid-js';
 
-const DayCardList: Component = () => {
-  return <div class="container mx-auto p-4 min-h-screen bg-main-bg"></div>;
+import type { DayType } from './types';
+
+// Components
+import DayCard from './day-card';
+import DetailedDayCard from './detailed-day-card';
+
+interface DayCardListProps {
+  list: DayType[];
+}
+
+const DayCardList: Component<DayCardListProps> = props => {
+  const [selectedCardId, setSelectedCardId] = createSignal<number>();
+  const [selectedCard, setSelectedCard] = createSignal<DayType | null>(null, {
+    equals: false,
+  });
+
+  createEffect(() => {
+    const selected = props.list.find(card => card.id === selectedCardId());
+    if (selected) {
+      setSelectedCard(selected);
+    } else {
+      setSelectedCard(null);
+    }
+  });
+
+  const isClicked = createSelector(selectedCardId);
+
+  return (
+    <>
+      <div class="flex gap-4 mb-2">
+        <For each={props.list}>
+          {(day, i) => (
+            <DayCard
+              day={day}
+              onCardClick={dayId => {
```

Comment:
wow. 

not sure, I understood the full idea.
could we just set card on click?

Reply:
could we do
`
const DayCardList: Component<DayCardListProps> = props => {
  const [selectedCard, setSelectedCard] = createSignal<DayType | null>(null, {
    equals: false,
  });

  const isClicked = createSelector(selectedCard);

  return (
    <>
      <div class="flex gap-4 mb-2">
        <For each={props.list}>
          {(day, i) => (
            <DayCard
              day={day}
              `
             onCardClick={card => {
                if (!selectedCard()) {
                  setSelectedCard(card);
                } else {
                  setSelectedCard(null);
                }
              }}
              isClicked={isClicked(day)}
            />
`
          )}
        </For>
      </div>

      <Show when={selectedCard()} keyed>
        {card => <DetailedDayCard day={card} />}
      </Show>
    </>
  );
};
`


// src/components/day-card-list/day-card.tsx

interface DayCardProps {
  day: DayType;
  `onCardClick: (day: DayType) => void;`
  isClicked: boolean;
}


.....
  return (
    <div
      class={
        props.isClicked
          ? `${cardStylesBasic} ${cardStylesHovered}`
          : cardStylesBasic
      }
      `onClick={() => props.onCardClick(props.day)}`
    >
.....

Reply:
Yes, absolutely. The idea was just to use createEffect to see how it works ))